python/fbprophet/diagnostics.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+import numpy as np
+import pandas as pd
+from functools import reduce
+from fbprophet import Prophet
+
+
+def _copy(model):
+    """Copy Prophet object
+
+    Parameters
+    ----------
+    model: Prophet class object.
+
+    Returns
+    -------
+    Prophet class object with the same parameter with model variable
+    """
+
+    # TODO(@teramonagi) this function should be moved into Prophet class as a method.
+    result = Prophet(
+        growth=model.growth,
+        n_changepoints=model.n_changepoints,
```

Comment:
There are two interfaces for changepoints: The `n_changepoints` input that is copied over here, but there is also the `changepoints` input, which allows the user to manually specify the locations of any potential changepoints. If `changepoints` is not None, then `n_changepoints` is ignored.

The correct thing to do here is to copy over both, but one subtlety is that we require the user-specified changepoints to fall within the history (see `Prophet.validate_inputs`). We thus need to filter `model.changepoints` to be just those that are <cutoff (if `model.changepoints` is not None). This should be straightforward by passing `cutoff` along to `_copy`. Making it an optional input would make sense if it is a method of Prophet.